### PR TITLE
Add endpoint to archive an organisation

### PR DESCRIPTION
**Pull request by klssmith**
The new endpoint checks that an org has no users, pending invited users,
or services (in any state). If it does, admin will display an error message
so that the platform admin user archiving the org can manually reassign
/ remove the team members and services.

Assuming that the org can be archived, we remove its email branding,
letter branding and domains, and set its state to `active=False`.

[Pivotal story](https://www.pivotaltracker.com/story/show/178519743)

### CODE REVIEW

File: app/dao/organisation_dao.py
Comment (idavidmcdonald) [inline app/dao/organisation_dao.py:111]: I think in here we may also want to call `organisation.name = get_archived_db_column_value(organisation.name)` which is the equivalent to what we do for the service name in `dao_archive_service`

It probably makes very little difference but I guess 
- it keeps things consistent
- if we ever did want to create a new organisation later with the same name as the old one, we won't be blocked by the unique constraint on the organisation name

**Review by idavidmcdonald — APPROVED** [review]
Great work. This all looks good, simple and meets the expectations of the story.

One small comment that I think we should probably do but it is pretty minor.

File: app/dao/organisation_dao.py
Comment (klssmith) [reply app/dao/organisation_dao.py:111]: My thinking behind not doing that originally was that you could have a organisation, for example "Camden Council", which gets archived. If someone then wants to create a new organisation called "Camden Council" in future, it would essentially be the same organisation, so we'd want to just unarchive the existing org instead of setting up a brand new one with the same name. That might work better for the usage data.

This is a different to services, where the names can be anything, so we want to release the name for another team to use.

I can make the change to do what you suggest though if you think that we wouldn't want to unarchive an organisation like that

Comment (idavidmcdonald) [reply app/dao/organisation_dao.py:111]: Your call on that one. There are pros and cons.

At the moment though we don't offer a way to unarchive without going in the DB so if that is the case, I'd encourage someone to make a new organisation from scratch as that can be done using the user interface

I'll leave you to decide what approach to take as either case doesn't matter too much

Comment (klssmith) [reply app/dao/organisation_dao.py:112]: I've updated this to change the org name

**Review by idavidmcdonald — APPROVED** [review]
Thanks for the change. Looks good!

**Pull request merged**
